clippy_lints/src/types.rs

```diff
@@ -679,6 +679,23 @@ declare_clippy_lint! {
     "cast to the same type, e.g. `x as i32` where `x: i32`"
 }
 
+/// **What it does:** Checks for casts function pointer to the numeric type.
+///
+/// **Why is this bad?** Cast pointer not to usize truncate value.
```

Comment:
What it does: "Checks for casts of a function pointer to a numeric type except \`usize\`."
Why is this bad? Casting a function pointer to something other than \`usize\` could truncate the address value.

This would be a little more precise, what this lint does.